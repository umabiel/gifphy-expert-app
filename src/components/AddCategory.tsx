import React, {
  ChangeEvent,
  Dispatch,
  FormEvent,
  Fragment,
  SetStateAction,
  useState,
} from "react";

import PropTypes from "prop-types";

interface IProps {
  setCategories: Dispatch<SetStateAction<string[]>>;
}

export const AddCategory = ({ setCategories }: IProps) => {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((animes) => [inputValue, ...animes]);
      setInputValue("");
    }
  };
  return (
    <Fragment>
      <form onSubmit={handleSubmit}>
        <input type="text" value={inputValue} onChange={handleInputChange} />
      </form>
    </Fragment>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
