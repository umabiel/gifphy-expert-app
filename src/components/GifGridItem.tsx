import { IGifs } from "../core/oop/interfaces/IGifs";

export const GifGridItem = ({ id, title, url }: IGifs) => {
  return (
    <div className="card">
      <p>{title}</p>
      <img src={url} alt={title} />
    </div>
  );
};
