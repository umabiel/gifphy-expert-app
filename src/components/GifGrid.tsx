import React, { Fragment } from "react";
import { IGifs } from "../core/oop/interfaces/IGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

interface IProps {
  category: string;
}

export const GifGrid = ({ category = "One Punch" }: IProps) => {
  // const initState: IGifs[] = [];
  // const [images, setImages] = useState(initState);
  const { data: images, loading } = useFetchGifs(category);

  return (
    <Fragment>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid">
        {images.map((x: IGifs) => (
          <GifGridItem key={x.id} {...x} />
        ))}
      </div>
    </Fragment>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
