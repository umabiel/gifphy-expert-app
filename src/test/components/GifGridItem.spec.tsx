import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas con componente <GifGridItem />", () => {
  const title = "Encabezado";
  const imagen = "http://localhost/img/1.png";

  const wrapper = shallow(<GifGridItem title={title} url={imagen} />);

  test("debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar el titulo en una etiqueta parrafo <p />", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe mostrar el src de la imagen y el texto alternativo", () => {
    const img = wrapper.find("img");
    const { src, alt } = img.props();
    expect(src).toBe(imagen);
    expect(alt).toBe(title);
  });
});
