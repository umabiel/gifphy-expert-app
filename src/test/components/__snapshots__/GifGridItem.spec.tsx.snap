// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con componente <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card"
>
  <p>
    Encabezado
  </p>
  <img
    alt="Encabezado"
    src="http://localhost/img/1.png"
  />
</div>
`;
